### PR TITLE
cache_promote plugin: put a mutex-protected critical section into a Continuation.

### PR DESCRIPTION
**Pull request by ywkaras**
This is a draft PR that takes the place of https://github.com/apache/trafficserver/pull/5406 .

### CODE REVIEW

**Comment by ywkaras** [comment]
[approve ci autest]

File: plugins/cache_promote/cache_promote.cc [hunk 1]
```diff
@@ -337,10 +286,128 @@ class LRUPolicy : public PromotionPolicy
...
+      TSDebug(PLUGIN_NAME, "LRUPolicy::doPromote(%.*s%s)", url_len > 100 ? 100 : url_len, url, url_len > 100 ? "..." : "");
+      _hash.init(url, url_len);
+      TSfree(url);
+
+      ContinueInMemberFunc<Do_promote_steps, &Do_promote_steps::critical_section>::once(*this, _policy->_lock)->schedule();
+    }
+
+  private:
+    TSHttpTxn _txnp;
+    LRUPolicy *_policy;
```
Comment (ywkaras) [inline plugins/cache_promote/cache_promote.cc]: I guess this also needs a dummy TSMutex to pass to the after_critical_section continuation.

**Comment by ywkaras** [comment]
[autest ci Clang-Analyzer]

**Comment by ywkaras** [comment]
[approve ci Clang-Analyzer]

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by ywkaras** [comment]
@bryancall  vetoed https://github.com/apache/trafficserver/pull/4848 , which this PR depends on, so this PR thus effectively vetoed also.

**Comment by bryancall** [comment]
@ywkaras I didn't veto it.  The procedure is to close old PRs that haven't had any recent activity. Please read the comment at the end of the PR.

**Comment by ywkaras** [comment]
I have no way of forcing meaningful activity so thus it is effectively vetoed.

**Pull request closed without merging**